src/js/services/history.js

```diff
@@ -48,7 +48,16 @@ export default App.extend({
   triggerChange() {
     this.getChannel().trigger('change:route', this._prevHistory[0]);
   },
-  goBack() {
+  goBack(onEmptyHistory) {
+    if (onEmptyHistory && this._prevHistory.length < 2) {
+      onEmptyHistory();
+      return;
+    }
+
+    // Remove the current route from the stack
+    this._prevHistory.shift();
```

Comment:
yeah so it's a _little_ awkward.. if you're directly linking to a form the `_prevHistory` will be just that form url.  so then the go back isn't go back in history.. it's just linking to the patient..  

so in theory if there was a "go back" button on the patient page, if we remove it here, that button wouldn't go back to the form, it would go to the default app route.  I think you'd want it to bounce back and forth between the two routes. 

The `onEmptyHistory` would be clearing the history instead of doing some sort of default redirect, and I don't know if that's the ideal behavior.